Examples/Detectors/Geant4Detector/include/ActsExamples/Geant4Detector/Geant4DetectorService.hpp

```diff
@@ -0,0 +1,90 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/ProtoDetector.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/Framework/BareService.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+#include "ActsExamples/Geant4/GdmlDetectorConstruction.hpp"
+
+namespace Acts {
+class TrackingGeometry;
+namespace Experimental {
+class Detector;
+}  // namespace Experimental
+}  // namespace Acts
+
+namespace ActsExamples {
+namespace Geant4 {
+
+/// @class Geant4DetectorService
+///
+class Geant4DetectorService final : public BareService {
```

Comment:
Actually, I'm not really sure why we inherit from `BareService` here... We do not plug it into the sequencer, right?
As far as I see the code would work the same without inheritance (only used locally in the `constructDetector`/`constructTrackingGeometry` functions), but maybe I overlook something.

Could we even make this a local class in the the `src` folder and only expose the config object?

Reply:
hmmm not sure. if it should work as a service the inheritance sounds fine. if it is just to get the logging mechanics IMO copy pasting the logging stuff would be better. but up to you. let me know how you chose and I will approve for merging @asalzburger

Reply:
I take your point, will adapt accordingly tomorrow.